Src/Extern/amrdata/AMReX_AmrData.cpp

```diff
@@ -877,7 +877,7 @@ bool AmrData::ReadNonPlotfileData(const string &filename, Amrvis::FileType /*fil
     dataGrids[LevelZero][ComponentZero]->define(fabBoxArray, dMap, NVarZero,
                                                 fabNGrow, Fab_noallocate);
     if(ParallelDescriptor::IOProcessor()) {
```

Comment:
Is AMReX supposed to be expcetion safe? `auto newfab = std::make_unique<FArrayBox>()` and following `newfab.get()` for non-owning accesses would be more c++ iodiomatic

Reply:
We never really care much about exception safety.  The problem is for us it's not clear how we could handle it if the memory allocation throws an exception other than crashing the code.  I am afraid of touching the code in AmrData.  So I just did the minimal change to it.

Reply:
We also have quite a lot of `unique_ptr::reset(new ...)` because we had been using C++11.  Some day, we need to replace it with `make_unique`.

Reply:
Makes sense. LGTM